lib/web_ui/dev/test_platform.dart

```diff
@@ -1059,7 +1059,18 @@ class BrowserManager {
         }
 
         _controllers.add(controller!);
-        return await controller!.suite;
+
+        final List<Future<RunnerSuite>> futures = <Future<RunnerSuite>>[
+          controller!.suite
+        ];
+        if (_browser.onUncaughtException != null) {
+          futures.add(_browser.onUncaughtException!.then<RunnerSuite>(
```

Comment:
I think I'd mildly prefer `_browser.onUncaughtException` to be a callback function. It would clean up this code. For example, you could simply do:

```
_browser.onUncaughtException = controller!.onError;
```

And then you wouldn't need to manage a list of futures.

That said, I don't know what kind of concurrency is going on here. Perhaps, the futures can already be completed by the time this code is reached, and so a callback will have no effect.

Reply:
This code is waiting for the completion of either:
* `controller!.suite` (which completes when the suite has been loaded)
* `_browser.onUncaughtException` (which completes if the browser sees an uncaught exception during loading)

If the suite loading future completes, then this function will return the suite.  If the exception future completes, then this function will throw the exception.